src/components/sidenav/sidenav.js

```diff
@@ -323,6 +326,77 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
       }
     }
 
+    function enableDragging() {
+      $mdGesture.register(element, 'drag', { horizontal: true });
+      element.on('$md.dragstart', onDragStart)
+        .on('$md.drag', onDrag)
+        .on('$md.dragend', onDragEnd);
+
+      var style = getComputedStyle(element[0]);
+      var sidenavWidth = parseInt(style.width);
+      var isRightSidenav = element.hasClass('md-sidenav-right');
+
+      var dragCancelled = false;
+      var dragPercentage;
+      var lastOpenState;
+
+      function onDragStart() {
+        if (element.hasClass('md-locked-open')) {
+          dragCancelled = true;
+        } else {
+          lastOpenState = scope.isOpen;
+          element.css($mdConstant.CSS.TRANSITION_DURATION, '0ms');
+        }
+      }
+
+      function onDrag(ev) {
+        if (dragCancelled) return;
+
+        // If the sidenav is aligned left, the sidenav is opened at zero x-axis
+        var startOffset = isRightSidenav ? 100 : 0;
+
+        dragPercentage = startOffset - Math.round((ev.pointer.distanceX / sidenavWidth) * 100);
+
+        if (dragPercentage > 100) dragPercentage = 100;
+        else if (dragPercentage < 0) dragPercentage = 0;
+
+        element.css($mdConstant.CSS.TRANSFORM, 'translate3d(-' + dragPercentage + '%,0,0)');
+      }
+
+      function onDragEnd() {
+        if (dragCancelled) {
+          dragCancelled = false;
+          return;
+        }
+
+        var remainingPercentage = dragPercentage > 50 ? 100 - dragPercentage : dragPercentage;
+        var animationTime = 4 * remainingPercentage;
```

Comment:
That's because of my percentage calculation
- https://github.com/angular/material/blob/eb1695a0d54f6794a08f04169ad8a390b92a1849/src/core/style/variables.scss#L95

`400ms / 100% = 4ms`

So every percentage needs to run for 4ms to reach a smooth transition as always.